src/main/java/uk/gov/hmcts/darts/common/client/multipart/StreamingMultipart.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.darts.common.client.multipart;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.darts.common.multipart.SizesableInputSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@Slf4j
+public class DefaultMultipart implements MultipartFile {
+
+    private final String name;
+
+    @Nullable
+    private final String contentType;
+
+    private final SizesableInputSource is;
+
+    private final InputStream stream;
+
+    public DefaultMultipart(
+        String name, @Nullable String contentType, SizesableInputSource source) throws IOException {
+        this.name = name;
+        this.contentType = contentType;
+        this.is = source;
+        stream = is.getInputStream();
+    }
+
+
+    @Override
+    public String getName() {
```

Comment:
don't suppose we can use lombok instead of this boilerplate code?

Reply:
yep we can. Done